docs: Add style guide (#2744)
opendp/opendp: docs/README.md

python3 -m venv .venv-docs
source .venv-docs/bin/activate
pip install -r requirements.txt
```

### Python docs

> Once depencies are installed, to buid python docs:

Reviewer: build

Reviewer: dependencies

Reviewer: Minor typos in this line

> Once depencies are installed, to buid python docs:

Author: ```suggestion
Once dependencies are installed, to build python docs:
```

```shell
make html
open build/html/index.html
```

This local build differs from the hosted docs, generated by `docs.yml`:

> - In CI we use spinx-multiversion to build docs for all releases of OpenDP.

Reviewer: sphinx*

> - In CI we use spinx-multiversion to build docs for all releases of OpenDP.

Author: ```suggestion
- In CI we use sphinx-multiversion to build docs for all releases of OpenDP.
```

- [API](https://docs.opendp.org/en/stable/api/index.html) is a reference primarily for Python OpenDP library users, and provides links to R and Rust documentation. It includes both a top-down "User Guide" and a bottom-up "API Reference".
- [Theory](https://docs.opendp.org/en/stable/theory/index.html) provides background information and motivation for differential privacy, and OpenDP in particular.
- [Contributing](https://docs.opendp.org/en/stable/contributing/index.html) is for people interested in contributing to the OpenDP Library.

After someone has read "Contributing", they can turn to a `README.md` or comments in the code for the details on a particular part of the codebase. 

### RST idioms

The OpenDP library is multilingual:
Although there are separate API references for R and Rust, the high-level descriptions are ideally applicable to multiple languages.

> We will use RST `tab-set` and `tab-item` for consistency, even if the only language documented is Python. Code is typically inline or but `literalinclude` is also used, particularly if we need a particular example in a stand-alone file for testing.

Reviewer: seems like a thought change that was left over inline [or] but

> We will use RST `tab-set` and `tab-item` for consistency, even if the only language documented is Python. Code is typically inline or but `literalinclude` is also used, particularly if we need a particular example in a stand-alone file for testing.

Author: ```suggestion
We will use RST `tab-set` and `tab-item` for consistency, even if the only language documented is Python. Code is typically inline, but `literalinclude` is also used, particularly if we need a particular example in a stand-alone file for testing.
```

Python code should be formatted as doctests in `.. code:: pycon` blocks, to ensure that the code continues to work as intended.

An isolated epsilon is easiest as a unicode character: `ε`. For more complex expressions, use `:math:` to inline LaTeX, or `.. math::` for a LaTeX block.

Respect the semantics of RST: Don't use a blockquote when `.. note::` is more appropriate.
Remember that RST is not Markdown:
```
For a ``function_name`` use double backticks.
Single asterisk is *italics*, and double is **bold**.
The `link syntax <https://example.com/>`_ trips me up,
but use :ref:`reference syntax <ref-syntax>` for internal links.
```

### Utility scripts and tests

We are gradually moving Jupyter notebook (`.ipynb`) examples to RST.
Doctests in RST give us a stronger check that the code is still functioning as intended, while notebooks are JSON internally, making diffs harder to read.
`tools/nb-to-rst.py` can make a partial translation, following our conventions for code formatting.

> In CI, we check the formatting of code in docs with `blacken-docs`, and `test_docs.py` has other ad hoc checks.

Reviewer: I think the ending seems disconnected from the and although it might be stylistic.

Author: I guess I was trying to point out that there is code that reads the documentation source, apart from the `tools` directory, but I don't want to get into the weeds either. Not sure what would be better.
